### DISCUSSION TITLE
Emitter - Bridging the imperative and the reactive worlds

### DISCUSSION THREAD

**Discussion by giscus** [quarkusio/quarkus#23568] [Quarkus Blog/Website]
# Emitter - Bridging the imperative and the reactive worlds

Quarkus: Supersonic Subatomic Java

https://quarkus.io/blog/reactive-messaging-emitter/

**Comment by fsomme2s** [comment]
How can I force the emitter to be synchronous / blocking? Running in a serverless context means I cannot wait for some background task to collect messages and send them "some point in the future" (because there is no future in serverless context)
...disclaimer: as far as I understand it now...

Actually when I try to use Emitter together with Quarkus for AWS Lambda, I get a "SRMSG00028: The subscription to mytopicname has been cancelled"

**Reply by josephearl** [reply]
+1

I thought using `OnOverflow.Strategy.NONE` (which then delegate to Mutiny `BackPressureStrategy.IGNORE`) would do this, but it doesn't (in my test case with this strategy messages get lost).

**Reply by bkalas** [reply]
Hi, if emitter emits to remote amqp broker,
you need to have in configuration ```mp.messaging.outgoing.<channelName>.* ``` configuration.
Could you explain ```failure-strategy``` configurations for outgoing channels?
Do outgoing channels use this attribute?  which of values ```fail (default), accept, release, reject, modified-failed``` are supported for outgoing?

**Reply by bkalas** [reply]
sorry should not be reply but new comment...

**Comment by bkalas** [comment]
Hi, if emitter emits to remote amqp broker,
you need to have in configuration mp.messaging.outgoing.<channelName>.* configuration.
Could you explain failure-strategy configurations for outgoing channels?
Do outgoing channels use this attribute? which of values fail (default), accept, release, reject, modified-failed are supported for outgoing?
